clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/config/BootstrapKubernetesConfigurationProvider.java

```diff
@@ -0,0 +1,148 @@
+ * are configured. So some of the features provided by SpringBoot and clouddriver modules do not
+ * work. Here are the limitations: 1. The properties in the yaml file must use camel case 2.
+ * Property placeholders don't work. 3. "encryptedFile:" and "encrypted:" notations don't work 4.
+ * "configserver:" works only for kubeconfig files
+ */
+public class BootstrapKubernetesConfigurationProvider implements KubernetesConfigurationProvider {
+
+  private final ConfigurableApplicationContext applicationContext;
+  private CloudConfigResourceService configResourceService;
+  private Map<String, String> configServerCache;
+
+  public BootstrapKubernetesConfigurationProvider(
+      ConfigurableApplicationContext applicationContext) {
+    this.applicationContext = applicationContext;
+  }
+
+  @Override
+  public KubernetesConfigurationProperties getKubernetesConfigurationProperties() {
+    return getKubernetesConfigurationProperties(getPropertiesMap());
+  }
+
+  @SuppressWarnings("unchecked")
+  public KubernetesConfigurationProperties getKubernetesConfigurationProperties(
+      Map<String, Object> kubernetesPropertiesMap) {
+    ObjectMapper objectMapper = new ObjectMapper();
+    // remove the keys having blank string values
+    kubernetesPropertiesMap.values().removeAll(Collections.singleton(""));
+
+    Map<String, Object> propertiesMap =
+        (Map<String, Object>)
+            JsonUnflattener.unflattenAsMap(kubernetesPropertiesMap).get("kubernetes");
+
+    KubernetesConfigurationProperties kubernetesConfigurationProperties =
+        objectMapper.convertValue(propertiesMap, KubernetesConfigurationProperties.class);
+
+    kubernetesConfigurationProperties
+        .getAccounts()
+        .forEach(
+            acc -> {
+              if (acc.getKubeconfigFile().startsWith("configserver:")) {
+                acc.setKubeconfigFile(resolveConfigServerFilePath(acc.getKubeconfigFile()));
+              }
+            });
+    return kubernetesConfigurationProperties;
+  }
+
+  @SuppressWarnings("unchecked")
+  private Map<String, Object> getPropertiesMap() {
+    ConfigurableEnvironment environment = applicationContext.getEnvironment();
+    Map<String, Object> map;
+
+    for (PropertySource<?> propertySource : environment.getPropertySources()) {
+
+      if (propertySource instanceof BootstrapPropertySource) {
+        map = (Map<String, Object>) propertySource.getSource();
+        return map;
+      }
+
+      if (propertySource.getSource() instanceof BootstrapPropertySource) {
+        BootstrapPropertySource<Map<String, Object>> bootstrapPropertySource =
+            (BootstrapPropertySource<Map<String, Object>>) propertySource.getSource();
+        return bootstrapPropertySource.getSource();
+      }
+    }
+
+    throw new RuntimeException("No BootstrapPropertySource found!!!!");
+  }
+
+  private String resolveConfigServerFilePath(String key) {
+    String filePath;
+
+    if (configResourceService == null) {
```

Comment:
Why are we manually fetching this bean here when we access it instead of autowiring as we do above for `ConfigurableApplicationContext applicationContext`?

Reply:
Now I autowired it.